File: Coding In Flow/Broadcast Receiver/GosyncReceiver/app/src/main/java/com/example/gosyncreceiver/MainActivity.java
package com.example.gosyncreceiver;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.text_view);
    }

    public void sendBroadcast(View view) {
        Intent intent = new Intent("com.example.EXAMPLE_ACTION");
        intent.setPackage("com.example.gosync");
        Bundle extras = new Bundle();
        extras.putString("stringExtra", "Start");
        sendOrderedBroadcast(intent, null, new SenderReceiver(),
                null, 0, "Start", extras);
    }
}

File: Personal/AidlDemo/aidl_client/src/main/java/com/vanphuc/aidl_client/MainActivity.kt
package com.vanphuc.aidl_client

import android.content.ComponentName
import android.content.Intent
import android.content.ServiceConnection
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.IBinder
import android.widget.Button
import com.vanphuc.aidldemo.MyAidlInterface

class MainActivity : AppCompatActivity() {
    var aidlColorService: MyAidlInterface? = null

    val connection = object : ServiceConnection {
        override fun onServiceConnected(p0: ComponentName?, p1: IBinder?) {
            aidlColorService = MyAidlInterface.Stub.asInterface(p1)
        }

        override fun onServiceDisconnected(p0: ComponentName?) {

        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val intent = Intent("AIDLColorServer").apply {
            setPackage("com.vanphuc.aidldemo")
        }

        bindService(intent, connection, BIND_AUTO_CREATE)

        val button = findViewById<Button>(R.id.button_color)
        button.setOnClickListener {
            val color = aidlColorService?.color
            it.setBackgroundColor(color!!)
        }
    }
}